ui/src/Pages/ArtistPage.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router";
+import NavBar from "../components/NavBar";
+import { Track, Artist } from "../types";
+import { Tracks } from "../components/Tracks";
+import PlayCircleRoundedIcon from "@mui/icons-material/PlayCircleRounded";
+import { green, red } from "@mui/material/colors";
+import FavoriteBorderOutlinedIcon from "@mui/icons-material/FavoriteBorderOutlined";
+import Auth from "../utils/Auth";
+export default function AlbumPage() {
+  const accessToken = Auth();
+  const { id } = useParams();
+  const [tracks, setTracks] = useState<Track[]>([]);
+  const [artist, setArtist] = useState({
+    name: "",
+    image: "",
+    followers: "",
+  });
+  const getArtist = () => {
+    const searchParameters = {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${accessToken}`,
+      },
+    };
+
+    fetch(`https://api.spotify.com/v1/artists/${id}`, searchParameters)
+      .then((response) => response.json())
+      .then((data: Artist) => {
+        setArtist({
+          name: data.name,
+          image: data.images[0].url,
+          followers: data.followers.total,
+        });
```

Comment:
Why do we need to separate out the data coming from the api?

Reply:
In this case, for Artist Page specifically, to get all necessary data, 3 API calls were needed which returned different kinds of data, therefore i separated the data into different use states.